core/src/main/java/com/sequenceiq/cloudbreak/converter/StackToTemplatePreparationObjectConverter.java

```diff
@@ -392,6 +394,14 @@ private boolean isCloudStorageConfigured(Stack source) {
 
     private GeneralClusterConfigs calculateGeneralClusterConfigs(Stack source, Cluster cluster) {
         GeneralClusterConfigs generalClusterConfigs = generalClusterConfigsProvider.generalClusterConfigs(source, cluster);
+        boolean allInstanceGroupsHaveMultiAz = source.getInstanceGroups().stream().allMatch(ig -> {
+            boolean instanceGroupHasMultiAz = ((List<String>) ig.getInstanceGroupNetwork()
+                    .getAttributes()
+                    .getMap()
+                    .getOrDefault(NetworkConstants.SUBNET_IDS, new ArrayList<>())).size() > 1;
+            return instanceGroupHasMultiAz;
```

Comment:
declaring the boolean var is not necessary here
also for clean coding reasons please extract the lambda body to a method